src/automata/include/automata/ata.hpp

```diff
@@ -130,12 +135,27 @@ AlternatingTimedAutomaton<LocationT, SymbolT>::AlternatingTimedAutomaton(
   const std::set<SymbolT> &                alphabet,
   const LocationT &                        initial_location,
   const std::set<LocationT> &              final_locations,
-  std::set<Transition<LocationT, SymbolT>> transitions)
+  std::set<Transition<LocationT, SymbolT>> transitions,
+  std::optional<LocationT>                 sink_location)
 : alphabet_(alphabet),
   initial_location_(initial_location),
   final_locations_(final_locations),
-  transitions_(std::move(transitions))
+  transitions_(std::move(transitions)),
+  sink_location_(sink_location)
 {
+	if (sink_location_) {
+		if (initial_location_ == *sink_location_) {
+			throw std::invalid_argument("The initial location must not be the sink location");
+		}
+		if (final_locations_.find(*sink_location_) != std::end(final_locations_)) {
+			throw std::invalid_argument("The sink location must not be a final location");
```

Comment:
I was wondering: If the sink location is a final location, that means that this is just a very complicated way of saying the language of the automaton is empty, right?